src/app/components/client-active-ride/client-active-ride.component.ts

```diff
@@ -15,20 +16,16 @@ import { RideService } from 'src/app/services/ride/ride.service';
 export class ClientActiveRideComponent implements OnInit {
 
   @Input() ride!: ActiveRide;
-  stops: Stop[] = [];
+  @Output() onRideCancelled = new EventEmitter<null>();
   currentEta: number = 0;
   msgs: Message[] = [];
+  client!: User;
+  private _stompClient!: Stomp.Client;
 
   constructor(private confirmationService: ConfirmationService, private rideService: RideService, private messageService: MessageService) {}
 
   ngOnInit(): void {
-    if (this.ride !== undefined) {
-      this.stops = this.ride.stops.map((s: Address) => {
-        const a = new AutocompleteAddress();
-        a.formatted_address = s.address;
-        return {address: a};
-      })
-    }
+    getSession() && this.initializeWebSocketConnection();
```

Comment:
what is this `getSession()` used for

Reply:
it is used as a guard clause.
if the user isn't logged in getSession() will return undefined and this.initializeWebSocketConnection() won't run.